test/QuadForm/Herm/Genus.jl

```diff
@@ -395,4 +395,28 @@
   sig = Dict(rlp[1] => 2, rlp[2] => 2)
   G = genus([g], sig)
   @test G == genus([g], [(rlp[1], 2), (rlp[2], 2)])
+
+
+  # partial test codes after fixing genus_generators
+  Qx, x = PolynomialRing(FlintQQ, "x")
+  f = x^2 - 2
+  K, a = NumberField(f, "a", cached = false)
+  Kt, t = PolynomialRing(K, "t")
+  g = t^2 + 1
+  E, b = NumberField(g, "b", cached = false)
+  D = matrix(E, 2, 2, [1, 2, 2, 1])
+  gens = Vector{Hecke.NfRelElem{nf_elem}}[map(E, [1, 0]), map(E, [a, 0]), map(E, [b + 1, 0]), map(E, [1//2*a*b + 1//2*a, 0]), map(E, [0, 1]), map(E, [0, a]), map(E, [0, b + 1]), map(E, [0, 1//2*a*b + 1//2*a])]
+
+  L = hermitian_lattice(E, generators = gens, gram_ambient_space = D)
+  gens, b, P0 = @inferred Hecke.genus_generators(L)
+  @test length(gens) == 0
+  @test b == false
+  @test P0 == Hecke.smallest_neighbour_prime(L)[2]
+
+  L = Hecke.HermLat(E, identity_matrix(E,8))
+  gens, b, P0 = @inferred Hecke.genus_generators(L)
+  @test length(gens) == 0
+  @test b == true
+  @test P0 == Hecke.smallest_neighbour_prime(L)[2]
```

Comment:
what is the rationale behind this test?

Reply:
I just wanted to make sure that the outputs of both functions stay coherent... If at some point we change the outputs of one function without making the other one crash, we still ensure that we have the correct outputs. It was also to make sure that we don't change the output `P0` of `smallest_neighbour_prime` while running `genus_generators` (I checked line by line but in case I missed one of them)

Reply:
Fair enough.